### PR TITLE
[RLlib] Fix KL method of MultiCategorial tf distribution (issue #7009).

### PR DESCRIPTION
**Pull request by sven1977**
The KL-method of tf-MultiCategorical was incorrect. It was not reducing properly (must leave the batch intact and reduce only over the n sub-categoricals).

Fixes issue #7009 

Closes issue #7009 

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: rllib/optimizers/async_replay_optimizer.py [hunk 1]
```diff
@@ -3,15 +3,16 @@
...
+import numpy as np
 import os
 import random
-import time
-import threading
-
-import numpy as np
 from six.moves import queue
+import threading
+import time
```
Comment (ericl) [inline rllib/optimizers/async_replay_optimizer.py]: Should remove the unrelated changes from this PR.

Comment (sven1977) [reply rllib/optimizers/async_replay_optimizer.py]: I just reordered the imports alphabetically (my PyCharm does that).

Comment (sven1977) [reply rllib/optimizers/async_replay_optimizer.py]: Ah, yes, different branch. Fixed.

**Comment by sven1977** [comment]
@ericl Jenkins all ok. Please merge.

**Review by ericl — APPROVED** [review]
